src/components/ToDo/ToDo.tsx

```diff
@@ -1,13 +1,109 @@
-import DateAndDay from "../DateAndDay/DateAndDay";
-import TaskList from "../TaskList/TaskList";
+import DateAndDay, { formattedDate } from "../DateAndDay/DateAndDay";
+import TaskList, { TaskProps } from "../TaskList/TaskList";
 import "./ToDo.css";
+import Input from "../Input/Input";
+import { ChangeEvent, useCallback, useEffect, useState } from "react";
+import { v4 as uuidv4 } from "uuid";
 
 function ToDo() {
+  const [inputShow, setInputShow] = useState(false);
+  const [valid, setValid] = useState(false);
+  const [enteredTask, setEnteredTask] = useState("");
```

Comment:
Having all states in root component will cause a performance problem.
Here, you are setting a state `enteredTask` on every single keystroke, It will rerender entire root component (ToDo) on every single character entered in input.
Check in profiler.